### ISSUE TITLE
Ability to append to console output

### ISSUE THREAD

**Issue by jbrad** [eslint/eslint#2840]
I was wondering if there is any interest in supporting the ability to append to the console output for different linters. I work for a large corporate/distributed team and it would be nice to add some additional information for failures. One idea is that different rules could have links to documentation for more information for specific rules. This can help with onboarding and educating people along with use for pull requests and code reviews.

I have been able to do this by creating custom linters that are essentially a copy with the appended string for output. Here's an example of what I'm talking about,

```js
if (!isValid) {
    context.report(node, "Strings must use " + settings.description + ". For me info, please checkout https://link.to.documentation");
}
```

There are some rules like `max_statements` and `complexity` where this makes most sense to me, I used strings as a quick example.

I think it would be nice to add an option in the `.eslintrc` file to pass this in via the configuration file for `rules`.

**Comment by ilyavolodin** [comment]
I believe somebody already made a reporter where rule names are clickable and lead to documentation. But I don't remember the name of it (and I think it only works on Macs).

**Comment by zpratt** [comment]
+1

**Comment by skawaguchi** [comment]
+1

**Comment by mysticatea** [comment]
I'm not sure we need this, if we do this, I think this should be a part of [plugin's config](http://eslint.org/docs/developer-guide/working-with-plugins).

e.g.:

```js
module.exports = {
    "rules": { },
    "docUri": "https://example.com/rules/{{name}}.md"
};
```

Embedded rules can resolve the document URI from only its rule name. So if plugins also have its document URI, we can treat those a common way.

**Comment by nzakas** [comment]
@jbrad assuming you mean custom formatters, not custom linters?

I'm not clear on what the request is here. I'm assuming you're not just asking for direct links to documentation, but rather a way to append arbitrary data to the end of output?

In both cases, I'd say a custom formatter is the way to go.

**Comment by btmills** [comment]
@ilyavolodin [royriojas/eslint-friendly-formatter](https://github.com/royriojas/eslint-friendly-formatter), and yeah, it requires iTerm2 specifically

**Comment by nzakas** [comment]
Closing, as I believe this should be handled with a custom formatter.
